src/autograd/Functions.cpp

```diff
@@ -147,6 +158,15 @@ namespace af {
             return Variable(result, {input}, grad_func);
         }
 
+        Variable relu(const Variable &input)
+        {
+	    auto result = max(input.array(), constant(0, input.array().dims()));
+	    auto grad_func = [](std::vector<Variable> &inputs, const Variable &grad_output) {
+	      inputs[0].addGrad(relu(grad_output) > 0);
+            };
+            return Variable(result, {input}, grad_func);
+        }
```

Comment:
I think this can be simplified like this:

```
Variable relu(const Variable &input)
          {
            auto mask = input > 0;
  	    auto result = mask.array() * input.array();
  	    auto grad_func = [](std::vector<Variable> &inputs, const Variable &grad_output) {
  	      inputs[0].addGrad(grad_output * inputs[1]);
              };
              return Variable(result, {input, mask}, grad_func);
          }
```